File: Task2/apz-pzpi-21-11-milennyi-ivan-task2/src/view_models/feed_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct FeedVM {
    id: u64,
    amount: u32,
    name: String,
    calories: u32,
    fat: u32,
    protein: u32,
    carbohydrates: u32,
    breed_name: String,
    sheep_count: u64,
}

File: Task4/apz-pzpi-21-11-milennyi-ivan-task4/src/viewModels/FeedVM.ts
export class FeedVM {
    id: number;
    amount: number;
    name: string;
    calories: number;
    fat: number;
    protein: number;
    carbohydrates: number;
    breedName: string;
    sheepCount: number;

    constructor(id: number, amount: number, name: string, calories: number, fat: number, protein: number, carbohydrates: number, breedName: string, sheepCount: number) {
        this.id = id;
        this.amount = amount / 1000;
        this.name = name;
        //Format later
        this.calories = calories;
        this.fat = fat;
        this.protein = protein;
        this.carbohydrates = carbohydrates;
        this.breedName = breedName;
        this.sheepCount = sheepCount;
    }
}